### PR TITLE
Lesson 7 Part 2: /sys and /proc interfaces

### PR DESCRIPTION
**Pull request by Noxybot**

### CODE REVIEW

File: 02-KernelModules_CharacterDevice/Create_device.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    ret = len;
+    goto out;
+    }
+    
+    
+
+    out:
+        printk(KERN_DEBUG "Wrote %d bytes, my buffer size now is: %d\n", len, data_size);
+        return ret;
+}
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/Create_device.h:94]: add empty line at the end of file

Comment (galkinletter) [reply 02-KernelModules_CharacterDevice/Create_device.h:94]: Create_device.h  it is bad name for *.h

File: 02-KernelModules_CharacterDevice/Makefile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+obj-m := Character_driver.o
+
+.PHONY: all clean
+
+all:
+	$(MAKE) -C ${KDIR} M=$$PWD modules
+
+clean:
+	$(MAKE) -C ${KDIR} M=$$PWD clean
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/Makefile:11]: Please, add empty line at the end of file

File: 02-KernelModules_CharacterDevice/Makefile [hunk 3]
```diff
@@ -0,0 +1,11 @@
+KDIR := /home/eduard/Kernel_i386
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/Makefile:1]: PATH  need fix
like 
PATH := $(PATH):/my/dir

File: 02-KernelModules_CharacterDevice/Memory_management.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+
+static void delete_buffer(void){
+    if(driver_buffer != NULL){
+        kfree(driver_buffer);
+        driver_buffer = NULL;
+    }
+    data_size = 0;
+}
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/Memory_management.h:31]: Please, add empty line at the end of file

File: 02-KernelModules_CharacterDevice/Sys_interface.h [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    }
+    return count;
+}
+static struct  kobject *kobj_dir;
+static struct kobj_attribute file_attr = __ATTR(SYS_FILE_NAME, S_IWUSR, NULL, store_clear);
+
+void init_sys(void){
+    kobj_dir = kobject_create_and_add(SYS_DIR_NAME, NULL);
+    sysfs_create_file(kobj_dir, &file_attr.attr);
+}
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/Sys_interface.h:24]: Please, add empty line at the end of file

**Review by galkinletter — CHANGES_REQUESTED** [review]
Need fix code style

File: 02-KernelModules_CharacterDevice/Create_device.h [hunk 6]
```diff
@@ -0,0 +1,94 @@
+#pragma once
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/Create_device.h:1]: Header files in the Linux kernel are used for two purposes:
    1. to define interfaces between components of the kernel, and
    2. to define interfaces between the kernel and user space 
Plz put function implementations into .c files.

**Review by oleg-h2o — CHANGES_REQUESTED** [review]
Plz move function implementations from .h files to .c

**Pull request closed without merging**